### ISSUE TITLE
Panic when two nodes in the same route added to AggregatedNode

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#758]
I was just playing around with aggregated nodes and noticed that if you try and add two nodes from the same route to an aggregated node it crashes Python with this c error:

```
glp_set_mat_row: i = 8; ind[2] = 1; duplicate column indices not allowed
Error detected in file ..\src\api\prob1.c at line 779
```

If you try and do this within a notebook, there is no indication about what went wrong. The kernel just cashes.

I know you probably shouldn't be doing this (or need to) but it might be worth putting a explicit warning in the doc string about it.

_Originally posted by @Batch21 in https://github.com/pywr/pywr/pull/756#issuecomment-523084921_

**Comment by snorfalorpagus** [comment]
There is no way to catch this kind of error from GLPK so we need to check for this condition in `setup()`.

**Comment by jetuk** [comment]
Does `glp_error_hook` help with catching these errors?

**Comment by snorfalorpagus** [comment]
`glp_error_hook` sounds like something we should investigate (#759). I also think we should be catching these errors specifically though.

**Comment by jetuk** [comment]
> I also think we should be catching these errors specifically though.

Agreed.

**Comment by jetuk** [comment]
@Batch21 what happens when you do the same test with the `glpk-edge` solver?

**Comment by Batch21** [comment]
There is no error when using `glpk-edge` For reference, I'm testing this with the aggreagted1.json model from the tests folder with an extra link node added:

``` 
"nodes": [
        {
            "name": "A",
            "type": "input",
            "max_flow": 50
        },
        {
            "name": "B",
            "type": "input",
            "max_flow": 50
        },
        {
            "name": "C",
            "type": "input",
            "max_flow": 50
        },
        {
            "name": "X",
            "type": "link",
            "max_flow": 50
        },
        {
            "name": "Z",
            "type": "output",
            "max_flow": 100,
            "cost": -100
        },
        {
            "name": "agg",
            "type": "AggregatedNode",
            "nodes": ["A", "B", "C", "X"],
            "factors": [1.0, 1.0, 1.0, 1.0],
            "flow_weights": [1.0, 1.0, 1.0, 1.0],
            "max_flow": 30.0,
            "min_flow": 5.0
        }
    ],
    "edges": [
        ["A", "X"],
        ["X", "Z"],
        ["B", "Z"],
        ["C", "Z"]
    ]

```

**Comment by jetuk** [comment]
> There is no error when using `glpk-edge`

@Batch21 what's the solution for `glpk-edge`? A, B, C & X flow = 7.5? I assume this only works because the factor for A and X are the same?

**Comment by Batch21** [comment]
yeah the solution is 7.5 for all the nodes. If the factors for A and X are different than `glpk-edge` returns a "no feasible solution exists" error. If the flow weights are different then `glpk-edge` runs succesfully. For example, if the flow weights for A, B, C, X are 2.0, 1.0, 1.0, 1.0 then the solution flow for all nodes is 6.0. I haven't had a chance to dig into the `glpk-edge` code, so I'm not really sure what is going on.

**Comment by jetuk** [comment]
> If the factors for A and X are different than glpk-edge returns a "no feasible solution exists" error

This is a graceful Python exception, and not an exception fault?

> For example, if the flow weights for A, B, C, X are 2.0, 1.0, 1.0, 1.0 then the solution flow for all nodes is 6.0

This makes sense to me. They are just factors to apply to ensure the `5.0 <= agg <= 30` constraint.
